src/asymmetric_crypto_ctx.cpp

```diff
@@ -401,6 +401,27 @@ namespace mococrw {
      * ###########
      */
 
+namespace {
+    ECDSASignatureFormat SDAtoDSASignatureFormat (ECSDASignatureFormat sigFormat) {
+        return static_cast<ECDSASignatureFormat>(static_cast<int>(sigFormat));
+    }
+
+    std::vector<uint8_t> _Asn1ECSignatureToIEEE1363EcSignature(const std::vector<uint8_t> &signature,
+                                                               size_t keySizeBytes) {
+        auto ecdsa = _d2i_ECDSA_SIG(signature);
+
+        auto r = _ECDSA_SIG_get0_r(ecdsa.get());
+        auto s = _ECDSA_SIG_get0_s(ecdsa.get());
+
+        std::vector<uint8_t> ieee(keySizeBytes*2);
+
+        BN_bn2binpad(r, ieee.data(), keySizeBytes);
```

Comment:
I guess Martin was already leaving a comment here.
From openssl doc: 
BN_bn2binpad() returns the number of bytes written or -1 if the supplied buffer is too small.

Can you please either check the return value or add a comment why this is not necessary here?
Surely you are free to add this function to mococrw using add_function.py :)

Reply:
yeah, rebase with tears :-D